File: src-tdoa/flask/tools/db_test.py
# This script inserts a new coordinate location into the db every 10th of a second.

import sqlite3
import time
import random

def create_connection(db_file):
    """ connect to db """
    conn = None
    try:
        conn = sqlite3.connect(db_file)
        return conn
    except sqlite3.Error as e:
        print(e)
    return conn

def insert_coordinate(conn, x, y):
    """ insert new coordinate into coordinates table """
    sql = ''' INSERT INTO coordinates(x, y)
              VALUES(?,?) '''
    cur = conn.cursor()
    cur.execute(sql, (x, y))
    conn.commit()
    return cur.lastrowid

def main():
    database = "../cid.db"

    # connect to db
    conn = create_connection(database)
    
    if conn is not None:
        try:
            while True:
                # gen random coordinates
                x = random.randint(0, 960)  # temp ranges for display area
                y = random.randint(0, 750)  

                # insert new oordinate
                insert_coordinate(conn, x, y)
                
                # print to console for confirmation
                print(f"Inserted coordinates ({x}, {y})")

                time.sleep(0.1)
        except KeyboardInterrupt:
            print("Stopped by user.")
        finally:
            conn.close()
    else:
        print("Error! Cannot create the database connection.")

if __name__ == '__main__':
    main()

File: src-tdoa/flask/tools/db_test.cpp
// g++ -o db_test db_test.cpp -lsqlite3
// This script inserts a new coordinate location into the db every 10th of a second.

#include <sqlite3.h>
#include <cstdlib>
#include <iostream>
#include <chrono>
#include <thread>
#include <random>

// connect to db
sqlite3* create_connection(const char* db_file) {
    sqlite3* db;
    int rc = sqlite3_open(db_file, &db);
    if (rc != SQLITE_OK) {
        std::cerr << "Error opening database: " << sqlite3_errmsg(db) << std::endl;
        return nullptr;
    }
    return db;
}

// insert a new coordinate into coordinates table
bool insert_coordinate(sqlite3* db, int x, int y) {
    sqlite3_stmt* stmt;
    const char* sql = "INSERT INTO coordinates(x, y) VALUES(?,?)";

    if (sqlite3_prepare_v2(db, sql, -1, &stmt, nullptr) != SQLITE_OK) {
        std::cerr << "Failed to prepare statement: " << sqlite3_errmsg(db) << std::endl;
        return false;
    }

    sqlite3_bind_int(stmt, 1, x);
    sqlite3_bind_int(stmt, 2, y);

    if (sqlite3_step(stmt) != SQLITE_DONE) {
        std::cerr << "Failed to execute statement: " << sqlite3_errmsg(db) << std::endl;
        sqlite3_finalize(stmt);
        return false;
    }

    sqlite3_finalize(stmt);
    return true;
}

int main() {
    const char* database = "../cid.db";

    // connect to db
    sqlite3* db = create_connection(database);
    if (!db) return -1;

    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<> distrib(0, 750);  // temp range for box on HTML page

    // insert coordinate into table every 10th of a second
    while (true) {
        int x = distrib(gen);
        int y = distrib(gen);

        if (!insert_coordinate(db, x, y)) {
            std::cerr << "Insert failed" << std::endl;
            break;
        }

        std::cout << "Inserted coordinates (" << x << ", " << y << ")" << std::endl;

        std::this_thread::sleep_for(std::chrono::milliseconds(100));
    }

    sqlite3_close(db);
    return 0;
}
